src/kubernetes_resources.py

```diff
@@ -0,0 +1,189 @@
+            raise ResourceHandlingFailed(
+                f"Failed to create a new ServiceAccount: {e}"
+            ) from e
+
+    def create_cluster_role():
+        """Create ClusterRole resource for training-operator."""
+        # Get k8s API instance once authenticated
+        api_instance = K8sResources._get_api_client(
+            kubernetes.client.RbacAuthorizationV1Api
+        )
+
+        namespace = K8sResources.namespace()
+        name = K8sResources.name
+
+        # Populate metadata, rules, and body parameters
+        metadata = kubernetes.client.V1ObjectMeta(name=name, labels={"app": name})
+        rule_1 = kubernetes.client.V1PolicyRule(
+            api_groups=["kubeflow.org"],
+            resources=[
+                "tfjobs",
+                "mxjobs",
+                "pytorchjobs",
+                "xgboostjobs",
+                "tfjobs/status",
+                "pytochjobs/status",
+                "mxjobs/status",
+                "xgboostjobs/status",
+            ],
+            verbs=["create", "delete", "get", "list", "patch", "update", "watch"],
+        )
+        rule_2 = kubernetes.client.V1PolicyRule(
+            api_groups=[""],
+            resources=["pods", "services", "endpoints", "events"],
+            verbs=["*"],
+        )
+        rule_3 = kubernetes.client.V1PolicyRule(
+            api_groups=["scheduling.volcano.sh"], resources=["podgroups"], verbs=["*"]
+        )
+        body = kubernetes.client.V1ClusterRole(
+            api_version="rbac.authorization.k8s.io/v1",
+            kind="ClusterRole",
+            metadata=metadata,
+            rules=[rule_1, rule_2, rule_3],
+        )
+        field_manager = "juju"
+
+        try:
+
+            # TODO: this stays alive after removing the charm, remove handler
+            # is needed to prevent this.
+            # Manually removing the existing one before creating a new one
+            api_instance.delete_cluster_role(name)
+            api_instance.create_cluster_role(body=body, field_manager=field_manager)
+        except kubernetes.client.exceptions.ApiException as e:
+            raise ResourceHandlingFailed(
+                f"Exception when calling RbacAuthorizationV1Api->create_cluster_role: {e}"
+            ) from e
+
+    def create_cluster_role_binding():
+        """Create ClusterRole resource for training-operator."""
+        # Get k8s API instance once authenticated
+        api_instance = K8sResources._get_api_client(
+            kubernetes.client.RbacAuthorizationV1Api
+        )
+
+        namespace = K8sResources.namespace()
+        name = K8sResources.name
+
+        # Populate metadata, role_ref, subjects, and body parameters
+        metadata = kubernetes.client.V1ObjectMeta(name=name, labels={"app": name})
+        role_ref = kubernetes.client.V1RoleRef(
+            api_group="rbac.authorization.k8s.io",
+            kind="ClusterRole",
+            name="training-operator",
```

Comment:
```suggestion
            name=name,
```

Reply:
I guess none of this applies anymore. Good catch, though.